File: java/src/vetores/SomaVetor.java
package vetores;

import java.util.Scanner;

/**
 * Problema "soma_vetor"
 * Faça um programa que leia N números reais e armazene-os em um vetor. Em seguida:
 * - Imprimir todos os elementos do vetor
 * - Mostrar na tela a soma e a média dos elementos do vetor
 */
public class SomaVetor {
  private static final int MIN_SIZE = 1;
  private static final int MAX_SIZE = 10;

  /**
   * The entry point of application.
   *
   * @param args the input arguments
   */
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);

    System.out.printf("Quantos números você vai digitar (máximo de %d)? ", MAX_SIZE);
    int n = sc.nextInt();

    n = n < MIN_SIZE || n > MAX_SIZE ? MIN_SIZE : n;

    double[] vector = new double[n];

    for (int i = 0; i < vector.length; i++) {
      System.out.printf("Digite um número (%d de %d): ", i + 1, n);
      vector[i] = sc.nextDouble();
    }

    double summation = 0.0;
    for (double v : vector) {
      summation += v;
    }

    final double average = summation / n;

    System.out.print("VALORES = ");

    for (double v : vector) {
      System.out.printf("%.1f ", v);
    }

    System.out.printf("%nSOMA = %.2f", summation);
    System.out.printf("%nMÉDIA = %.2f%n", average);

    sc.close();
  }
}

File: python/src/vetores/soma_vetor.py
""" Problema "soma_vetor"
Faça um programa que leia N números reais e armazene-os em um vetor.
Em seguida:
 - Imprimir todos os elementos do vetor
 - Mostrar na tela a soma e a média dos elementos do vetor
"""

MIN_SIZE = 1
MAX_SIZE = 10


def soma_vetor() -> None:
    n = int(input(f"Quantos números você vai digitar (mínimo de {MIN_SIZE} e máximo de {MAX_SIZE})? "))

    if not MIN_SIZE < n <= MAX_SIZE:
        n = MIN_SIZE

    vector: [float] = n * [0.0]

    for i in range(len(vector)):
        vector[i] = float(input(f"Digite um número ({i + 1} de {len(vector)}): "))

    summation = 0.0
    for value in vector:
        summation += value

    average = summation / len(vector)

    print("VALORES =", end="")

    for value in vector:
        print(f" {value:.1f}", end="")

    print(f"\nSOMA = {summation:.2f}")
    print(f"MÉDIA = {average:.2f}")


if __name__ == "__main__":
    soma_vetor()
